Flaky test: App1Test.earlyEofTest

`App1Test.earlyEofTest` is regularly failing on Windows with Java 14 but not in other combinations.

Running the build again most of the time succeeds but is very annoying.

```
[ERROR] Failures: 
[ERROR]   App1Test.earlyEofTest:82 
Expecting:
 <false>
to be equal to:
 <true>
but was not.
```

--- a/dropwizard-dependencies/pom.xml
+++ b/dropwizard-dependencies/pom.xml
@@ -64,6 +64,7 @@
 
         <!-- Test dependencies -->
         <assertj.version>3.16.1</assertj.version>
+        <awaitility.version>4.0.3</awaitility.version>
         <bcprov-jdk15on.version>1.66</bcprov-jdk15on.version>
         <h2.version>1.4.200</h2.version>
         <hsqldb.version>2.5.1</hsqldb.version>
@@ -492,6 +493,11 @@
                 <artifactId>bcprov-jdk15on</artifactId>
                 <version>${bcprov-jdk15on.version}</version>
                 <scope>test</scope>
+            </dependency>
+            <dependency>
+                <groupId>org.awaitility</groupId>
+                <artifactId>awaitility</artifactId>
+                <version>${awaitility.version}</version>
             </dependency>
         </dependencies>
     </dependencyManagement>

--- a/dropwizard-e2e/pom.xml
+++ b/dropwizard-e2e/pom.xml
@@ -132,6 +132,11 @@
             <artifactId>assertj-core</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.awaitility</groupId>
+            <artifactId>awaitility</artifactId>
+            <scope>test</scope>
+        </dependency>
     </dependencies>
 
     <build>

--- a/dropwizard-e2e/src/test/java/com/example/app1/App1Test.java
+++ b/dropwizard-e2e/src/test/java/com/example/app1/App1Test.java
@@ -8,7 +8,6 @@ import io.dropwizard.testing.ResourceHelpers;
 import io.dropwizard.testing.junit5.DropwizardAppExtension;
 import io.dropwizard.testing.junit5.DropwizardExtensionsSupport;
 import io.dropwizard.util.Duration;
-
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
@@ -23,9 +22,11 @@ import java.net.URL;
 import java.nio.charset.StandardCharsets;
 import java.util.Collections;
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.entry;
+import static org.awaitility.Awaitility.await;
 
 @ExtendWith(DropwizardExtensionsSupport.class)
 public class App1Test {
@@ -77,9 +78,9 @@ public class App1Test {
         conn.getOutputStream().write("{".getBytes(StandardCharsets.UTF_8));
         conn.disconnect();
 
-        // Wait a bit for the app to process the request.
-        Thread.sleep(500);
-        assertThat(((App1)RULE.getApplication()).wasEofExceptionHit).isTrue();
+        await().atMost(5, TimeUnit.SECONDS)
+                .until(() -> ((App1) RULE.getApplication()).wasEofExceptionHit);
+        assertThat(((App1) RULE.getApplication()).wasEofExceptionHit).isTrue();
     }
 
     @Test

